src/tests/ftest/pool/management_race.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import time
+import random
+from apricot import TestWithServers
+from thread_manager import ThreadManager
+from command_utils_base import CommandFailure
+
+class PoolManagementRace(TestWithServers):
+    """
+    Epic: Create system level tests that cover pool management race boundary tests and
+          functionality.
+    Testcase:
+          DAOS-8463: Pool management integrity with race boundary conditions
+    Test Class Description:
+          Start DAOS servers, create pools and delete the pool while query and list.
+    :avocado: recursive
+    """
+
+    def del_recreate_query_and_list_pools(
+            self, del_recreate=True, pool_num=1, thread_num=1, test_loop=2):
+        """ To delete and recreate pool with the same label, to query and list pools.
+        Args:
+            del_recreate (bool): flag to delete and recreate pool with the same label.
+            pool_num (int): pool number.
+            thread_num (int): thread number.
+            test_loop (int): number of test loop to be executed.
+        """
+
+        wait_time = self.params.get("wait_time", '/run/boundary_test/*')
+        max_query_time = self.params.get("max_query_time", '/run/boundary_test/*')
+        pool_label_prefix = self.params.get("pool_label_prefix", '/run/boundary_test/*')
+        label = pool_label_prefix + str(pool_num)
```

Comment:
```suggestion
        label = self.pool[pool_num].label
```
I think the best way to track and randomly select the pool label is to use the default label. When you create 20 pools, in default, each pool will get a label like `TestLabel_1`, `TestLabel_2`, and so on. Then you can randomly select the pool and destroy it and recreate with that label.

Reply:
If we use the default label, then the new label is generated incrementally after pool deleted and recreated (ie. Testlabel_21, TestLabel22...). There will be less chance to trigger the race condition for query_threads.
So, it is intentional to use the same pool label after re-created for the multiple query_threads.

Reply:
You can set whatever the label you want to the new pool. e.g.,
```
# Get random index.
index = random.randint(0, len(self.pool)-1)
# Save the label for recreation.
label = self.pool[index].label
# Destroy the pool.
self.pool[index].destroy()
# Remove the pool from self.pool
del self.pool[index]
# Create a new pool with the same label.
self.pool.append(self.get_pool(create=False))
# Reuse the label.
self.pool[-1].label.update(label)
# Create the pool.
self.pool[-1].create()
```
Anyway, just a suggestion.

Reply:
I agree with Makito. We should simplify as possible. I think this also works
```
# Destroy
self.pool[index].destroy()
# Recreate, which uses the same label
self.pool[index].create()
```
Then, we could just use the default label names an simplify a lot of the label logic in this file.

Reply:
Right. This will make it much simpler.

Reply:
self.pool[index].create() will generate a incremental label.
We can not use the default label here, since the query_thread doesn't know the newly created incremental label to query and trigger the race condition.

Reply:
`get_pool()` calls `get_params()` and `create()`:
https://github.com/daos-stack/daos/blob/38a74edf846ec909b1f82172fe2dca6b209a0288/src/tests/ftest/util/apricot/apricot/test.py#L1611-L1619

`get_params()` generates the label:
https://github.com/daos-stack/daos/blob/38a74edf846ec909b1f82172fe2dca6b209a0288/src/tests/ftest/util/test_utils_pool.py#L135-L140

`create()` uses the already generated label: https://github.com/daos-stack/daos/blob/38a74edf846ec909b1f82172fe2dca6b209a0288/src/tests/ftest/util/test_utils_pool.py#L235-L244

So you can do:

```
# Create a pool with auto-generated label
self.pool.append(self.get_pool())

# Destroy the pool
self.pool[-1].destroy()

# Re-create the pool - uses the same label
self.pool[-1].create()
```

Reply:
Please see the new commit.